x/staking/simulation/operations.go

```diff
@@ -413,8 +413,19 @@ func SimulateMsgCancelUnbondingDelegate(ak types.AccountKeeper, bk types.BankKee
 			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCancelUnbondingDelegation, "account does have any unbonding delegation"), nil, nil
 		}
 
-		// get random unbonding delegation entry at block height
-		unbondingDelegationEntry := unbondingDelegation.Entries[r.Intn(len(unbondingDelegation.Entries))]
+		// this is a temporary fix to make staking simulation pass.
+		// we should fetch the first unbondingDelegationEntry that matches the creationHeight.
+		// because currently the staking msgServer chooses the first unbondingDelegationEntry with the matching creationHeight.
```

Comment:
```suggestion
		// This is a temporary fix to make staking simulation pass. We should fetch
		// the first unbondingDelegationEntry that matches the creationHeight, because
		// currently the staking msgServer chooses the first unbondingDelegationEntry
		// with the matching creationHeight.
		//
		// ref: https://github.com/cosmos/cosmos-sdk/issues/12932
```